main.js

```diff
@@ -0,0 +1,319 @@
+            return firstNumber - secondNumber;
+        }
+    );
+}
+
+console.log(getSmallestNumber([2, -5, 10, 1, 4])); // -5
+console.log(getSmallestNumber([200, 25, 4, 123, 87])); // 4
+
+// 3
+
+function getSquaredNumbers(numbersArray) {
+    let squaredNumbers = [];
+
+    for (let i = 0; i < numbersArray.length; i++) {
+        squaredNumbers.push(numbersArray[i] * numbersArray[i]);
+    }
+    return squaredNumbers;
+}
+
+getSquaredNumbers([1, 2, 3, 4, 5]); // [1, 4, 9, 16, 25]
+getSquaredNumbers([6, 7, 8, 9, 10]); // [36, 49, 64, 81, 100]
+
+const numbers = [1, 2, 3];
+const squaredNumbers = getSquaredNumbers(numbers);
+console.log(squaredNumbers); // [1, 4, 9]
+console.log(numbers); // [1, 2, 3]
+console.log(numbers !== squaredNumbers); // true
+
+// 4
+
+function getReversedString(string) {
+    const stringSplitByCharacters = string.split('');
+    return stringSplitByCharacters.reverse();
+}
+
+console.log(getReversedString('Hello!'));
+console.log(getReversedString('Arrays'));
+
+// 5
+
+function isPalindrome(text) {
+    const textWithoutSpecialCharacters = text.replace(/[^a-z]/gi, '').toLowerCase();
+    const reversedText = textWithoutSpecialCharacters.split('').reverse().join('');
+
+    return textWithoutSpecialCharacters === reversedText;
+}
+
+console.log(isPalindrome('Kayak'));
+console.log(isPalindrome('Hello!'));
+console.log(isPalindrome('Was it a cat I saw'));
+
+// 6
+
+function countLetters(string) {
+    const letterCount = {};
+    const splittedAndCleanedString = string.replace(/[^a-zA-Z]/g, '').toLowerCase().split('');
+
+    for (i = 0; i < splittedAndCleanedString.length; i++) {
+        const letter = splittedAndCleanedString[i];
+
+        if (letterCount[letter] > 0) {
+            letterCount[letter] = letterCount[letter] + 1;
+        } else {
+            letterCount[letter] = 1;
+        }
+    }
+    return letterCount;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7
+
+const peopleArray = [
+    {
+        name: 'Adam',
+        age: 20
+    },
+    {
+        name: 'Amanda',
+        age: 5
+    },
+    {
+        name: 'John',
+        age: 75
+    },
+    {
+        name: 'Dave',
+        age: 15
+    }
+]
+
+function getYoungestPerson(people) {
+   let youngestPerson = people[0].age;
+
+   for (i = 1; i < people.length; i++) {
+       if (people[i].age < youngestPerson) {
+           return youngestPerson = people[i].age;
+       }
+   }
+   return youngestPerson;
+}
+
+function getOldestPerson(people) {
+    let oldestPerson = people[0].age;
+
+    for (i = 1; i < people.length; i++) {
+        if (people[i].age > oldestPerson) {
+            return oldestPerson = people[i].age;
+        }
+    }
+    return oldestPerson;
+}
+
+function getAgeDifference(people) {
+    const oldestPersonAge = getOldestPerson(peopleArray);
+    const youngestPersonAge = getYoungestPerson(peopleArray);
```

Comment:
Those functions are called `getOldestPerson` and `getYoungestPerson`, not `getOldestPersonAge` and `getYoungestPersonAge`. That's a **huge** difference. Do you know why?